### PR TITLE
Add backend implementation

### PR DESCRIPTION
**Pull request by tpazderka**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,6 +1,8 @@
 [isort]
 line_length = 120
 add_imports = from __future__ import unicode_literals
+known_third_party = mozilla_django_oidc
```
Comment (ziima) [inline setup.cfg]: `default_section = THIRDPARTY` usually works better.

File: tox.ini [hunk 2]
```diff
@@ -1,15 +1,31 @@
 [tox]
-envlist = quality
+envlist =
+    quality
+    py{27,34,35,36}-django111
+    py{34,35,36}-django200
```
Comment (ziima) [inline tox.ini]: `django20` would match the `django111` better (bugfix number not included).

File: tox.ini [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
 passenv =
     PYTHONWARNINGS
+setenv =
+    DJANGO_SETTINGS_MODULE = django_oidc_user.tests.settings
+deps =
+    coverage
+    django111: Django>=1.11,<2.0.0
+    django200: Django>=2.0.0
+commands =
+    coverage run --source=django_oidc_user --branch -m django test django_oidc_user
```
Comment (ziima) [inline tox.ini]: codecov?

File: django_oidc_user/tests/settings.py [hunk 4]
```diff
@@ -7,4 +7,12 @@
     'django.contrib.contenttypes',
     'django.contrib.auth',
     'django_oidc_user',
+    'mozilla_django_oidc',
 ]
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': 'test',
```
Comment (ziima) [inline django_oidc_user/tests/settings.py]: `NAME = ':memory:'` would put the database into memory, which should be fastest.

File: tox.ini [hunk 5]
```diff
@@ -1,15 +1,37 @@
...
+    CI
+    TRAVIS
+    TRAVIS_*
+setenv =
+    DJANGO_SETTINGS_MODULE = django_oidc_user.tests.settings
+deps =
+    coverage
+    codecov
+    django111: Django>=1.11,<2.0.0
+    django200: Django>=2.0.0
```
Comment (ziima) [inline tox.ini]: Mismatch between `django20` and `django200`

**Comment by tpazderka** [comment]
Fixed.

**Pull request merged**